### DISCUSSION TITLE
Sechste Woche

### DISCUSSION THREAD

**Discussion by PetrusQn** [MelonigeMelone/xCalendar#16] [Blogkanal]
# Woche 6

## Implementieren einer weiteren Seite

In der vergangenen Woche haben wir eine neue "**Registrieren-Seite**" implementiert. Im folgenden Screenshot ist zu erkennen, dass wir (aufgrund des überwiegend positiven Feedbacks) am bereits vorhandenen Design festgehalten haben. 

![image](https://github.com/MelonigeMelone/xCalendar/assets/136441045/5d30f9eb-d183-43b3-aa6b-db17a3919924)

Nun ist es möglich zwischen den beiden Seiten (Login- und Registrieren-Seite) hin und her zu navigieren, was wir nicht mittels einfachen Links und Weiterleitungen, sondern via Angular-Routing-Feature realisiert haben. 

**Neben der Navigation ist bislang nur das Frontend, der jeweiligen Seite, implementiert**

## Login Demo 

Zu Demozwecken haben wir implementiert, dass sich ein Benutzer über die Login-Seite anmelden kann. Aktuell gibt es keinerlei Überprüfung der eingegebenen Daten, außer die Validierung der E-Mail (ob ein "@" darin vorkommt). 
Hat der User eine (im obigen Sinne) valide E-Mail-Adresse und ein dazu gehörendes Passwort (aktuell vollkommen frei wählbar) eingegeben und auf den Login-Button geklickt, so wird er zur exemplarischen "Homepage" weitergeleitet. Zu Demonstration kann dieser sich hier nochmals seine eingegebenen Daten ansehen. 

Bei Login:
![image](https://github.com/MelonigeMelone/xCalendar/assets/136441045/ca68b058-0487-4b31-87b2-6cae3ab93e1c)

Nach klicken des Login-Button:
![image](https://github.com/MelonigeMelone/xCalendar/assets/136441045/22fb446a-14f5-4a01-86ce-a6714f421a6e)

**Wie bereits erwähnt dient dies lediglich der Demonstration!**

Dennoch zur Umsetzung:
Durch den Klick auf den Login-Button werden die Benutzerdaten (E-Mail & Passwort) eingelesen. Parallel wird die E-Mail-Adresse auf Richtigkeit geprüft (ist "@" enthalten?). Beide Felder sind Pflichtfelder. Sie dürfen also nicht leer sein. Sind sie es dennoch, passiert beim Klick auf "Log in" nichts. Es wird lediglich angezeigt, dass die E-Mail eine Pflichtangabe ist. Wurden beide Felder korrekt ausgefüllt und der Login-Button angeklickt, werden die eingegebenen Daten mittels URL an die nächste Seite übermittelt, sodass diese auch dort verwendet werden können. 

## Setup zum Verwenden von "GraphQL"

Um die Kommunikation zwischen Client und Server zu gewährleisten, haben wir uns für "GraphQL" entschieden. Es ist eine flexible und performante Abfragesprache für eine API.

Das Einrichten von GraphQL in unserem Projekt war nicht so einfach wie wenn man es in Java einbinden würde. Erstmals musste eine "apollo-angular"-Bibliothek zum Projekt hinzugefügt werden. Daraufhin wurden man von der Konsole aufgefordert eine URL zu unserer GraphQL-API einzugeben und dann die Version von GraphQL, die verwendet werden soll. Die zugehörigen "Packages", wie sie in Angular genannt werden, wurden installiert.

![grafik](https://github.com/MelonigeMelone/xCalendar/assets/78598174/9cde7371-510a-4b39-8828-b1f5efcabcbb)

Außerdem wurde ein GraphQL-Modul in unserem Projekt automatisch generiert.

![grafik](https://github.com/MelonigeMelone/xCalendar/assets/78598174/5e01790c-d768-4bcb-8775-c0ed58a3dcd2)

Damit waren die erste Schritte getan und als nächstes sollten dann Modelle erstellt werden. Auf diesen Schritt kommen wir erst dann zurück, wenn wir eine Datenbank mit Daten, die wir auch abfragen können, zur Verfügung haben.

**Comment by Christian-2003** [comment]
Hallo zusammen,

es scheint, als hättet ihr diese Woche produktiv gearbeitet und viel geschafft.

Die Demo sieht sehr schön aus. Es sieht so aus, als hättet ihr Material Design für die Login-Seiten verwendet. Das gefällt mir gut. Wenn dem tatsächlich so ist, dann nehme ich an, dass ihr Material Design auch weiterhin für euren Kalender verwenden werdet?

Ich denke, dass ihr eine gute Entscheidung getroffen habt, GraphQL zu verwenden. Die Flexibilität wird sich bestimmt später auszahlen. Haltet uns diesbezüglich gerne auf dem Laufenden.

Insgesamt ist das ein sehr gelungener Blogeintrag. Ich freue mich auf nächste Woche.

**Comment by Joh4nta** [comment]
Zunächst war ich kurz verwirrt, weil hier Woche sechs steht und auf dem zugehörigen Aufgabenblatt Woche 7. 

ich finde es super, dass ihr bei der Beschreibung der Registrieren-Seite so detailliert auf die "Funktionen" und Eigenschaften eingegangen seid, die (noch) nicht implementiert sind. Wie z.B. das Prüfen des Passwortes. So ist für euch und uns klar, was noch gemacht werden kann / wird. 

Ansonsten sieht die Registrieren-Seite gut aus und es ist toll, dass auch eine Demo davon möglich ist. 

Es ist schön wie ausführlich ihr beschrieben habt, wie ihr das Setup für GraphQL eingerichtet habt.

**Comment by hmi95** [comment]
Danke für das Teilen der Erfahrungen beim Einbinden von GraphQL.
Bezüglich der Nummerierung war ich ähnlich verwirrt wie @JohannaRieschl
